grpc/src/main/java/com/linecorp/armeria/server/grpc/DefaultUnframedGrpcErrorHandler.java

```diff
@@ -0,0 +1,276 @@
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.Any;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.rpc.BadRequest;
+import com.google.rpc.DebugInfo;
+import com.google.rpc.ErrorInfo;
+import com.google.rpc.Help;
+import com.google.rpc.LocalizedMessage;
+import com.google.rpc.PreconditionFailure;
+import com.google.rpc.QuotaFailure;
+import com.google.rpc.RequestInfo;
+import com.google.rpc.ResourceInfo;
+import com.google.rpc.RetryInfo;
+
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.grpc.protocol.GrpcHeaderNames;
+import com.linecorp.armeria.common.logging.RequestLogAccess;
+import com.linecorp.armeria.common.logging.RequestLogProperty;
+import com.linecorp.armeria.common.util.Exceptions;
+import com.linecorp.armeria.internal.common.JacksonUtil;
+import com.linecorp.armeria.internal.common.util.TemporaryThreadLocals;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import io.grpc.Status;
+import io.grpc.Status.Code;
+
+final class DefaultUnframedGrpcErrorHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultUnframedGrpcErrorHandler.class);
+
+    private static final MessageMarshaller ERROR_DETAILS_MARSHALLER =
+            MessageMarshaller.builder()
+                             .omittingInsignificantWhitespace(true)
+                             .register(RetryInfo.getDefaultInstance())
+                             .register(ErrorInfo.getDefaultInstance())
+                             .register(QuotaFailure.getDefaultInstance())
+                             .register(DebugInfo.getDefaultInstance())
+                             .register(PreconditionFailure.getDefaultInstance())
+                             .register(BadRequest.getDefaultInstance())
+                             .register(RequestInfo.getDefaultInstance())
+                             .register(ResourceInfo.getDefaultInstance())
+                             .register(Help.getDefaultInstance())
+                             .register(LocalizedMessage.getDefaultInstance())
+                             .build();
+
+    private static final ObjectMapper mapper = JacksonUtil.newDefaultObjectMapper();
+
+    /**
+     * Ensure that unframedGrpcStatusMappingFunction never returns null by falling back to the default.
+     */
+    private static UnframedGrpcStatusMappingFunction ofStatusMappingFunction(
```

Comment:
Could you place the private static methods(or static methods that are called from this class and test calsses) at the bottom of this class? It's a sort of convention. 😉

Reply:
From [organize](https://armeria.dev/community/developer-guide/#organize), the top-down order is like 
```
static fields
static methods
member fields
constructors
member methods
utility methods (both static and member)
inner classes
```
Should i move the static method to the bottom? 😉

Reply:
private static methods and static methods with `@VisibleForTesing` are utility methods so we prefer to put them at the bottom as the doc describes. 😄